.github/workflows/build.yaml

```diff
@@ -46,10 +64,55 @@ jobs:
         PYTHON: python
     - name: Install dependencies
       run: make setup
+    - name: Create changelog and bump
+      if: ${{ github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON }}
+      run: |
+        git config --global user.name "github-actions"
+        git config --global user.email "github-actions@github.com"
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        mkdir -p ${{ env.ARTIFACT_DIR }}
+        cz changelog --dry-run --incremental --unreleased-version "Release Notes" > ${{ env.ARTIFACT_DIR }}/CHANGELOG.md
+        # Update the CHANGELOG.md.
+        cz changelog
+        cz bump
     - name: Build the package
-      # We don't need to check and test the package separately
-      # because `make dist` runs those targets first and only
-      # builds the package if they succeed.
-      run: make dist
-    - name: Build the docs
-      run: make docs
+      # We don't need to check and test the package separately because `make dist` runs
+      # those targets first and only builds the package if they succeed.
+      run: |
+        make dist requirements sbom
+    - name: Compute package hash
+      if: ${{ matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON }}
+      id: compute-hash
+      shell: bash
+      run: |
+        set -euo pipefail
+        TARBALL_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*.tar.gz")
+        WHEEL_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*.whl")
+        SBOM_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*-sbom.json")
+        HTML_DOCS_PATH=$(find ${{ env.ARTIFACT_DIR }} -name *-docs-html.zip)
+        CHANGELOG_PATH=$(find ${{ env.ARTIFACT_DIR }} -name CHANGELOG.md)
+        if [ $CHANGELOG_PATH -eq "" ];
+        then
+          touch ${{ env.ARTIFACT_DIR }}/CHANGELOG.md
+          CHANGELOG_PATH=${{ env.ARTIFACT_DIR }}/CHANGELOG.md
+        fi
```

Comment:
If the date of that `CHANGELOG.md` (the release notes, right?) doesn’t matter then you’d safe the check and just run
```
touch ${{ env.ARTIFACT_DIR }}/CHANGELOG.md
CHANGELOG_PATH=$(find ${{ env.ARTIFACT_DIR }} -name CHANGELOG.md)
```

Reply:
@jenstroeger see commit [3180dca](https://github.com/jenstroeger/python-package-template/pull/259/commits/3180dcab4f114ec74bf96124e00c0b60a8b12898) for the new changes.